core/src/wsv.rs

```diff
@@ -73,9 +74,22 @@ pub struct World {
     pub(crate) triggers: TriggerSet,
     /// Runtime Validator
     pub(crate) validator: Option<Validator>,
-    /// New version of Validator, which will replace `validator` on the next
-    /// [`validator_view()`](WorldStateView::validator_view) call.
-    pub(crate) upgraded_validator: Option<Validator>,
+}
+
+#[cfg(test)]
+impl Default for World {
+    fn default() -> Self {
+        Self {
+            parameters: Parameters::default(),
+            trusted_peers_ids: PeersIds::default(),
+            domains: DomainsMap::default(),
+            roles: crate::RolesMap::default(),
+            account_permission_tokens: crate::PermissionTokensMap::default(),
+            permission_token_definitions: crate::PermissionTokenDefinitionsMap::default(),
+            triggers: TriggerSet::default(),
+            validator: Some(Validator),
```

Comment:
Oh, I keep receiving this question everytime.

The problem is that we have unit-tests in `core`. At the same time `Validator` should be supplied by user at runtime (in genesis or in `Upgrade` instruction). Without `validator` set to `Some` unit-tests become kind of useless (and failing), because every transaction in tests requires validation. So we have a `MockValidator` which is aliased to `Validator` when `#[cfg(test)]`. It preforms no validation, but just executes provided instructions and queries (remember that *Validator* is in fact *Executor* and we will rename it in future).

I'll add a comment in this code.

Reply:
@Arjentix makes sense, I'm okay with that if there's a comment.
I like @Erigara's idea, but that's out of scope for this PR of course.

Reply:
Created #3752

Reply:
@Arjentix docs